test/e2e/volume_provisioning.go

```diff
@@ -125,6 +125,38 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 
 			testDynamicProvisioning(c, claim)
 		})
+
+		It("should test that deleting a claim before the volume is provisioned deletes the volume. [Slow][Volume]", func() {
+			// This case tests for the regressions of a bug fixed by PR #21268
+			// REGRESSION: Deleting the PVC before the PV is provisioned can result in PV
```

Comment:
Agreed.  Jeff and I actually hit on this yesterday also.  I've made some changes to iteratively create 100 PVCs then List PVs and check for any matching the pseudorandom storage class name.  Will push today after some testing / tweaking.